src/lib/transformers/tokend.js

```diff
@@ -96,22 +70,30 @@ class TokendTransformer {
    * @return {Promise<Properties>} properties that had $tokend values transformed
    */
   transform(properties) {
+    const seenProperties = [];
     const promises = collectTransformables(properties, []).map((info) => {
       const keyPath = info.get('keyPath');
+      if (keyPath.length !== 1) {
+        Log.log('WARN', 'keyPath does not have a length of 1, this could cause errors in the property set')
+      }
+
+      const propertyName = keyPath[0];
```

Comment:
What if `keyPath` is `null`, `undefined`, or something that's not an array with 1 element?

Reply:
ahh good catch, will update the if condition above to check, maybe we should throw an error if for this? When I talked to you before I believe we said logging should be fine?

Reply:
I think the `!== 1` might be a little aggressive. There may be a valid case where it returns an array with a length of 2 but we want the first element regardless.

My thought is definitely `reject` if `keyPath` isn't an array or is an array with `length == 0`. Otherwise, if `length > 1`, warn.

I feel like that covers all possible cases. What do you think?

Reply:
yeah, my `!== 1` is too aggressive and it failed some tests. I am going to do a little bit of debugging / comparison between master and this branch based on these tests just to make sure I am not missing anything drastic.